### PR TITLE
feat(sample): allow to save in IndexedDB

### PR DESCRIPTION
**Pull request by maasencioh**
An example of it usage

```js
var Sample = await API.require('vh/eln/Sample');
if (typeof IframeBridge !== 'undefined') {
    IframeBridge.onMessage(onMessage);
    IframeBridge.ready();
    function onMessage(data) {
        if (data.type === 'tab.data') {
            var sample = new Sample(data.message.couchDB, data.message.uuid, {track: true});
            API.cache('sample', sample);
            doPrediction();
        }
    }
} else {
    var {extSample} = await API.require('vh/eln/SampleFactory');
    var sample = await extSample({trackId: 'defaultSample'});
    sample.sample.triggerChange()
    API.cache('sample', sample);
    doPrediction();
}
```

### CODE REVIEW

File: eln/SampleFactory.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Sample from './ExtSample';
+
+export async function SampleFactory(options = {}) {
```
Comment (stropitek) [inline eln/SampleFactory.js]: I'd export `extSample` (lower case and distinct from possibly later added factory function for `Sample` with roc)

File: eln/ExtSample.js [hunk 2]
```diff
@@ -2,77 +2,106 @@ import Datas from 'src/main/datas';
...
+        };
 
-            this.bindChange();
+        this.bindChange();
+    }
+
+    async _init() {
+        this._initialized = new Promise(async (resolve) => {
+            if (this.options.uuid) {
+                var sample = await idb.get(this.options.uuid);
```
Comment (stropitek) [inline eln/ExtSample.js]: I would not call the option `uuid`. Maybe `trackId`?

**Comment by maasencioh** [comment]
@stropitek what do you think?

**Comment by targos** [comment]
Please make sure it ignores the error when IDB calls fail. There are issues with Firefox and IDB.

**Comment by maasencioh** [comment]
@targos I was not sure if I should ignore the set, because it is a promise inside the event callback, so it's not clear if that would be necessary, what do you think then?

**Comment by targos** [comment]
If the promise never resolves, that could be a problem

**Pull request merged**